include/swift/AST/LocalizationFormat.h

```diff
@@ -140,12 +140,10 @@ class SerializedLocalizationWriter {
 
 class LocalizationProducer {
 public:
-  /// If the  message isn't available/localized in the current `yaml` file,
-  /// return the fallback default message.
-  virtual llvm::StringRef getMessageOr(swift::DiagID id,
-                                       llvm::StringRef defaultMessage) const {
-    return defaultMessage;
-  }
+  /// Get a diagnostic  message from the current `yaml` or `.db` file. If it's
+  /// not available return `llvm::None`.
+  virtual llvm::Optional<llvm::StringRef>
+  getMessageOr(swift::DiagID id) const = 0;
```

Comment:
Nit: since there is no `default` anymore, this method should be renamed to just `getMessage`